dependency_manager/src/edm_tool/edm.py

```diff
@@ -624,11 +624,11 @@ def write_cmake(cls, workspace: dict, checkout: list, dependencies: dict, out_fi
             out.write(render)
 
 
-def checkout_local_dependency(name: str, git: str, git_tag: str, checkout_dir: Path) -> dict:
+def checkout_local_dependency(name: str, git: str, git_tag: str, checkout_dir: Path, keep_branch=False) -> dict:
     """
     Clone local dependency into checkout_dir.
 
-    If the directory already exists only switch branches if the git repo is not dirty.
+    If the directory already exists only switch branches if the git repo is not dirty or keep_branch is True
```

Comment:
keep_branch is False should be correct comment, I guess

Reply:
Fixed in the latest version.